app/models/invite_request.rb

```diff
@@ -1,8 +1,9 @@
 class InviteRequest < ApplicationRecord
   validates :email, presence: true, email_format: true
   validates :email, uniqueness: { message: "is already part of our queue." }
-  before_validation :compare_with_users, on: :create
   before_validation :set_simplified_email, on: :create
+  before_validation :check_admin_banned_list, on: :create
```

Comment:
It'd be better to make this a `validate` (I guess you copied compare_with_users, but that shouldn't be `before_validation` either...)